src/gen/mod.rs

```diff
@@ -0,0 +1,325 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use camino::Utf8Path;
+
+use genco::fmt;
+use genco::prelude::*;
+use serde::{Deserialize, Serialize};
+// use uniffi_bindgen::MergeWith;
+use crate::gen::oracle::DartCodeOracle;
+use uniffi_bindgen::{BindingGenerator, BindingsConfig, ComponentInterface};
+use self::render::Renderer;
+use self::types::TypeHelpersRenderer;
+
+mod compounds;
+mod enums;
+mod functions;
+mod objects;
+mod oracle;
+mod primitives;
+mod records;
+mod render;
+mod types;
+mod utils;
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct Config {
+    package_name: Option<String>,
+    cdylib_name: Option<String>,
+    #[serde(default)]
+    external_packages: HashMap<String, String>,
+}
+
+// impl MergeWith for Config {
+//     fn merge_with(&self, other: &Self) -> Self {
+//         let package_name = if other.package_name.is_some() {
+//             other.package_name.clone()
+//         } else {
+//             self.package_name.clone()
+//         };
+//         let cdylib_name = if other.cdylib_name.is_some() {
+//             other.cdylib_name.clone()
+//         } else {
+//             self.cdylib_name.clone()
+//         };
+//         Self {
+//             package_name,
+//             cdylib_name,
+//         }
+//     }
+// }
+
+impl From<&ComponentInterface> for Config {
+    fn from(ci: &ComponentInterface) -> Self {
+        Config {
+            package_name: Some(ci.namespace().to_owned()),
+            cdylib_name: Some(ci.namespace().to_owned()),
+            external_packages: HashMap::new(),
+        }
+    }
+}
+
+impl Config {
+    pub fn package_name(&self) -> String {
+        if let Some(package_name) = &self.package_name {
+            package_name.clone()
+        } else {
+            "uniffi".into()
+        }
```

Comment:
As I am just seeing it:
```suggestion
        self.package_name.clone().unwrap_or_else(||"uniffi".into())
```